### PR TITLE
Add warning in case open files limit is lower than recommended

### PR DESCRIPTION
**Pull request by zvolin**
Warn user if limit of open files for node's process is lower than recommended on unix systems.

image

### CODE REVIEW

File: cli/src/commands/start.rs [hunk 1]
```diff
@@ -27,6 +27,19 @@ use rand_chacha::ChaChaRng;
...
+/// Beacon should be able to handle at least 1000 concurrent connections,
+/// each requiring 2 sockets.
+#[cfg(target_family = "unix")]
+const RECOMMENDED_MIN_NOFILES_LIMIT_BEACON: u64 = 2048;
+/// Recommended minimum value of 'open files' limit for validator node.
+///
+/// Beacon should be able to handle at least 500 concurrent connections,
+/// each requiring 2 sockets.
+#[cfg(target_family = "unix")]
+const RECOMMENDED_MIN_NOFILES_LIMIT_VALIDATOR: u64 = 1024;
```
Comment (zvolin) [inline cli/src/commands/start.rs]: @howardwu @niklaslong please help choosing correct values for those constants :)

**Review by niklaslong — APPROVED** [review]
LGTM, though I'll defer to @howardwu regarding those limits.

File: cli/Cargo.toml [hunk 2]
```diff
@@ -30,6 +30,9 @@ features = ["derive"]
 [dependencies.colored]
 version = "2"
 
+[dependencies.nix]
```
Comment (howardwu) [inline cli/Cargo.toml]: We should cfg guard this dependency

Comment (zvolin) [reply cli/Cargo.toml]: good catch! I've just done it

just a note, as all checks have passed then maybe it would be good to add windows build to the ci?

**Review by howardwu — APPROVED** [review]
LGTM

**Comment by howardwu** [comment]
Verified CI failure is from an unrelated issue (likely outage)

Merging

**Pull request merged**